### ISSUE TITLE
gcc 10's -Werror=enum-conversion breaks compilation

### ISSUE THREAD

**Issue by AeroNotix** [litespeedtech/lsquic#148]
Hi,

Following the instructions in the README, results in this project being unable to compile:

```
[  0%] Building C object src/liblsquic/CMakeFiles/lsquic.dir/lsquic_full_conn.c.o
lsquic_full_conn.c: In function ‘generate_stop_waiting_frame’:
lsquic_full_conn.c:2780:59: error: implicit conversion from ‘enum <anonymous>’ to ‘enum packno_bits’ [-Werror=enum-conversion]
 2780 |     packnum_len = conn->fc_conn.cn_pf->pf_packno_bits2len(GQUIC_PACKNO_LEN_1);
      |                                                           ^~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [src/liblsquic/CMakeFiles/lsquic.dir/build.make:412: src/liblsquic/CMakeFiles/lsquic.dir/lsquic_full_conn.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:493: src/liblsquic/CMakeFiles/lsquic.dir/all] Error 2
make: *** [Makefile:114: all] Error 2
```

**Comment by dtikhonov** [comment]
Yikes, that's a new complaint. What compiler is this?

(And just remove `-Werror` from _CMakeLists.txt_ for now.)

**Comment by AeroNotix** [comment]
I am using `gcc`.

**Comment by AeroNotix** [comment]
Removing `-Werror` from CMakeLists.txt allows it to compile. Feel free to close if you think that's good enough. Seems like it should compile without removing that, though.

**Comment by dtikhonov** [comment]
I agree, it should compile out of the box...

What version of gcc are you using?

**Comment by AeroNotix** [comment]
10.2.0

**Comment by litespeedtech** [comment]
Fixed in 2.19.10: `-Werror` has been dropped.
